### PR TITLE
add new `connectorBaseImage` field to metadata

### LINKED ISSUE #30233: Declare a new metadata field: `connectorBaseImage`
Connector will declare which base image they should be built with through a new metadata field: `connectorBaseImage`. 
This field will reference an actual docker image version that should be existing in our dagger code and on DockerHub.

```yaml
data:
  allowedHosts:
    hosts:
      - api.openweathermap.org
  registries:
    oss:
      enabled: true
    cloud:
      enabled: false
  connectorBaseImage: airbyte/python-connector-base-image:0.1.0
  connectorSubtype: api
  connectorType: source
  definitionId: 561d7787-b45e-4f3b-af58-0163c3ba9d5a
  dockerImageTag: 0.2.0
  dockerRepository: airbyte/source-openweather
  githubIssueLabel: source-openweather
  icon: openweather.svg
  license: MIT
  name: Openweather
  releaseStage: alpha
  supportLevel: community
  documentationUrl: https://docs.airbyte.com/integrations/sources/openweather
  tags:
    - language:lowcode
metadataSpecVersion: "1.0"
```

This field should be optional as we'll still support builds from Dockerfile. The validation of the field value will happen later, when the first base image will actually be released.

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/30233
We want to define a new  field in the `metadata.yaml` to make connector determine the base image they're build with.
This field will be used in the build context of a connector to target the correct airbyte base image to use to build the connector.

## How
Declare a new top level data field: `build`. It's an object with a `baseImage` property.
Additional validation on `baseImage` can be considered in the future to check the docker image  exists (#30234).  

## 🚨 User Impact 🚨
None, this new field is optional.

### CODE REVIEW

**Comment by alafanechere** [comment]
test comment, my dispatched command should comment on it

File: airbyte-ci/connectors/metadata_service/lib/tests/fixtures/metadata_validate/valid/metadata_connector_base_image.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+data:
+  allowedHosts:
+    hosts:
+      - "*.googleapis.com"
+  connectorBaseImage: airbyte/airbyte-python-connectors-base:0.1.0
```
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/tests/fixtures/metadata_validate/valid/metadata_connector_base_image.yaml:5]: @alafanechere What are your thoughts around placing these types of fields under a `build` sections

```
data:
    build:
        connectorBaseImage: airbyte/airbyte-python-connectors-base:0.1.0
```

Comment (alafanechere) [reply airbyte-ci/connectors/metadata_service/lib/tests/fixtures/metadata_validate/valid/metadata_connector_base_image.yaml]: @bnchrch I made the change. I find using the field name `baseImage` more concise.

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Build.yaml
Comment (bnchrch) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/Build.yaml:1]: nit: Lets give it a more descriptive file name: e.g. `ConnectorBuildOptions.yaml`

**Review by bnchrch — APPROVED** [review]
Nice work!

**Pull request merged**